mobile_verifier/src/boosting_oracles/mod.rs

```diff
@@ -1,10 +1,305 @@
+        let handle = tokio::spawn(self.run(shutdown));
+        Box::pin(
+            handle
+                .map_err(anyhow::Error::from)
+                .and_then(|result| async move { result.map_err(anyhow::Error::from) }),
+        )
+    }
+}
+
+impl DataSetDownloaderDaemon {
+    pub fn new(
+        pool: PgPool,
+        urbanization: Arc<Mutex<Urbanization<DiskTreeMap, Geofence>>>,
+        store: FileStore,
+        oracle_boosting_sink: FileSinkClient,
+    ) -> Self {
+        Self {
+            pool,
+            urbanization,
+            store,
+            oracle_boosting_sink,
+        }
+    }
+
+    pub async fn run(self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        // Another option I considered instead of polling was to use ENOTIFY, but that seemed
+        // not as good, as it would hog a pool connection.
+        //
+        // We set the poll duration to 30 minutes since that seems fine.
+        let poll_duration = Duration::minutes(30);
+
+        loop {
+            // Find the latest urbanization file
+            tracing::info!("Checking for new data sets");
+            let mut urbanization = self.urbanization.lock().await;
+            let curr_urbanization_data_set = urbanization.timestamp();
+            let latest_data_set: Option<NewDataSet> = sqlx::query_as(
+                "SELECT file_name, time_to_use, status FROM data_sets WHERE status != 'processed' AND data_set = 'urbanization' AND COALESCE(time_to_use > $1, TRUE) AND time_to_use <= $2 ORDER BY time_to_use DESC LIMIT 1"
+            )
+                .bind(curr_urbanization_data_set)
+                .bind(Utc::now())
+                .fetch_optional(&self.pool)
+                .await?;
+
+            if let Some(latest_data_set) = latest_data_set {
+                let path = format!(
+                    "urbanization.{}.res10.h3tree",
+                    latest_data_set.time_to_use.timestamp()
+                );
+
+                // Download the file if it hasn't been downloaded already:
+                if !latest_data_set.status.is_downloaded() {
+                    tracing::info!("Downloading new data set: {path}");
+                    // TODO: abstract this out to a function
+                    let mut bytes = self.store.get(latest_data_set.file_name.clone()).await?;
+                    let mut file = File::open(Path::new(&path)).await?;
+                    while let Some(bytes) = bytes.next().await.transpose()? {
+                        file.write_all(&bytes).await?;
+                    }
+                    // Set the status to be downloaded
+                    sqlx::query("UPDATE data_sets SET status = 'downloaded' WHERE file_name = $1")
+                        .bind(&latest_data_set.file_name)
+                        .execute(&self.pool)
+                        .await?;
+                }
+
+                // Now that we've downloaded the file, load it into the urbanization data set
```

Comment:
Should all DB writes from this point forward not be wrapped in a txn ?  Commit that at same time as the filestore commit